test/functional/normalization/eventrouter_test.go

```diff
@@ -0,0 +1,118 @@
+package normalization
+
+import (
+	"encoding/json"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	"github.com/openshift/cluster-logging-operator/test/matchers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/reference"
+	"strings"
+	"time"
+)
+
+var (
+	templateForAnyKubernetes = types.Kubernetes{
+		ContainerName:     "*",
+		PodName:           "*",
+		NamespaceName:     "*",
+		NamespaceID:       "*",
+		OrphanedNamespace: "*",
+	}
+	templateForAnyCollector = types.PipelineMetadata{Collector: types.Collector{
+		Ipaddr4:    "*",
+		Inputname:  "*",
+		Name:       "*",
+		Version:    "*",
+		ReceivedAt: time.Time{},
+	},
+	}
+)
+
+var _ = Describe("[Normalization] Fluentd normalization for EventRouter messages", func() {
+
+	const timestamp string = "1985-10-21T09:00:00.00000+00:00"
+	var (
+		framework *functional.FluentdFunctionalFramework
+		pod       *corev1.Pod
+		nanoTime, _ = time.Parse(time.RFC3339Nano, timestamp)
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewFluentdFunctionalFramework()
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInput(logging.InputNameApplication).
+			ToFluentForwardOutput()
+		Expect(framework.Deploy()).To(BeNil())
+		pod = types.NewMockPod()
+	})
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	It("Should parse EventRouter `ADDED` message and check values", func() {
+		podRef, err := reference.GetReference(scheme.Scheme, pod)
```

Comment:
There is duplicate code here from each "It" that can extracted into a function to condense the work.  You might even consider adding it in `var` as a named function of some kind.  Given the similarity for all of these I wonder if you can do something like:

```
Describe("my test", func(){
  for _, verb := range []string{"ADDED","UPDATED"} {
    It(fmt.Sprintf("should parse %s", verb), func(){
       //testcode
    })
  }
})
```